sdk/core/core-http/src/serviceClient.ts

```diff
@@ -802,7 +817,7 @@ export function createPipelineFromOptions(
     tracingPolicy({ userAgent: userAgentValue }),
     keepAlivePolicy(keepAliveOptions),
     userAgentPolicy({ value: userAgentValue }),
-    generateClientRequestIdPolicy(),
+    ...(clientRequestIdPolicy ? [clientRequestIdPolicy] : []),
```

Comment:
feels weird - why are we putting it into an array just to spread it into a push call?